File: frontend/src/types/deal.ts
export type UserDealStatus = "saved" | "acquired" | "dismissed";

export interface SaveDealRequest {
  status: UserDealStatus;
  notes?: string | null;
}

export interface UserDealSummary {
  id: string;
  listingId?: string;
  status: UserDealStatus;
  notes?: string | null;
  createdAt: string;
  updatedAt?: string;
}

export interface UserDealResponse {
  id: string;
  listingId: string;
  status: UserDealStatus;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
}

File: backend/app/schemas/deal.py
from datetime import datetime

from app.schemas.base import CamelModel
from app.schemas.common import UserDealStatus
from app.schemas.opportunity import OpportunitySummaryResponse, UserDealSummaryResponse


class SaveDealRequest(CamelModel):
    status: UserDealStatus = "saved"
    notes: str | None = None


class UserDealResponse(CamelModel):
    id: str
    listing_id: str
    status: UserDealStatus
    notes: str | None = None
    created_at: datetime
    updated_at: datetime


class SavedDealItemResponse(CamelModel):
    user_deal: UserDealSummaryResponse
    opportunity: OpportunitySummaryResponse


class SavedDealListResponse(CamelModel):
    items: list[SavedDealItemResponse]
